testsuite/classlibrary/TestDocument.sc

```diff
@@ -0,0 +1,53 @@
+TestDocument : UnitTest {
+	// autorun doesn't have access to this object's scope
+	// need a public place to communicate the result
+	classvar <>success;
+
+	test_new_document_runs_initAction {
+		var success = false, doc, save;
+		if(Platform.ideName == "scqt") {
+			save = Document.initAction;
+			protect {
+				Document.initAction = { success = true };
+				doc = Document.new;
+				0.1.wait;
```

Comment:
`0.1.wait` in a test is a code smell. can you justify why this is necessary?

Reply:
Normal way to handle asynchronicity is with a `condition.hang` in the async response (initAction here). But we are testing whether the initAction fired or not. Therefore there is some reasonable expectation that the initAction may fail to fire. In that case, the Condition never unhangs and voilà, the entire test suite has ground to a halt.

This situation is impossible to test without a timeout. So, please advise what is the interface provided by the UnitTest suite for asynchronicity with timeout, and I'll very happily update (as I've been waiting years for that interface).

Reply:
BTW I've started working on a Condition:hangWithTimeout so that this might not be a problem in the future.

Reply:
i don't work a lot with UnitTest, but i think it's `UnitTest:wait`? if that doesn't work, leaving in the `0.1.wait` is fine as long as there is a comment justifying it for anyone looking at the code.

Reply:
Oh, but there is #3895. Well, what to do. I'll email the dev list.

Reply:
could you try `UnitTest:wait` and see if it works here?

Reply:
Yeah, UnitTest:wait is better. I didn't know about that one.

Reply:
BTW fixed 10 days ago. AFAICS this is ready to go?